### PR TITLE
Fix memory leak in TCP

### PR DESCRIPTION
**Pull request by gargnupur**
Signed-off-by: gargnupur <gargnupur@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: Fix memory leak in TCP
Ref: https://github.com/istio/istio/issues/22007
Risk Level: Low
Testing: Tested e2e in istio/proxy
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: source/extensions/common/wasm/context.cc [hunk 1]
```diff
@@ -281,6 +281,16 @@ WasmResult Context::setTickPeriod(std::chrono::milliseconds tick_period) {
   return WasmResult::Ok;
 }
 
+void Context::onCloseTCP() {
+  if (tcp_connection_closed_) {
+    return;
+  }
+  tcp_connection_closed_ = true;
+  onDone();
```
Comment (PiotrSikora) [inline source/extensions/common/wasm/context.cc]: We should check the result of `onDone` and postpone `onLog` and `onDelete` until other extensions agree, shouldn't we, @jplevyak?

Comment (jplevyak) [reply source/extensions/common/wasm/context.cc]: That logic is specifically for RootContexts and is already handled elsewhere.  The stream contexts are required to return "true" from the onDone call (the SDK ensure that)..   It is expected that any long lived operations which could outlast the stream will be handled by the RootContext.

Comment (gargnupur) [reply source/extensions/common/wasm/context.cc]: So, in that case.. no need to handle result of onDone.. Guys, if you agree with the PR, please approve and merge it. Don't have merge rights for envoy-wasm...

**Pull request merged**